src/indexers/vault-harvests-indexer.ts

```diff
@@ -1,19 +1,45 @@
 import { getDataMapper, getVaultEntityId } from '../aws/dynamodb.utils';
 import { VaultYieldEvent } from '../aws/models/vault-yield-event.model';
 import { getSupportedChains } from '../chains/chains.utils';
-import { loadYieldEvents, queryLastHarvestBlock } from '../vaults/harvests.utils';
+import { HARVEST_SCAN_BLOCK_INCREMENT, loadYieldEvents, queryLastHarvestBlock } from '../vaults/harvests.utils';
+import { YieldEvent } from '../vaults/interfaces/yield-event';
 
 export async function updateVaultHarvests() {
   const chains = getSupportedChains();
   const mapper = getDataMapper();
 
   for (const chain of chains) {
+    const sdk = await chain.getSdk();
+    const currentBlock = await sdk.provider.getBlockNumber();
     const vaults = await chain.vaults.all();
 
     for (const vault of vaults) {
       try {
-        const lastHarvestBlock = await queryLastHarvestBlock(chain, vault);
-        const yieldEvents = await loadYieldEvents(chain, vault, lastHarvestBlock);
+        let lastHarvestBlock = await queryLastHarvestBlock(chain, vault);
+        let yieldEvents: YieldEvent[] = [];
+
+        console.log(`[${vault.name}]: Last Indexed Block ${lastHarvestBlock}`);
+        while (true) {
```

Comment:
any way to do it without a while(true)

Reply:
well, unless you have a suggestion i don't really have an issue with a while true that is provably non perma looping. 
i believe it would be trivial to write a proof to show this loop will always terminate

Reply:
other guards like checking for no entries unfortunately do not work so i'm not sure of a good alternative given that the only inputs we have are block number, and whether or not we found entries, and lack of entries does not guarantee we are done unless we are out of blocks to check

Reply:
i guess this already has timeouts in lamda?

Reply:
yes correct lambda will enforce a timeout - that is the issue with this one i mention that we will need to capture the created block into the graph to 

- speed up indexes in general 
- ensure we do not run into an issue down the road of a vault constantly timing out bc its 11m start block and it was deployed like 3 months from now on block 18m or smth